compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs

```diff
@@ -1136,7 +1145,13 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                             } else {
                                 "".to_string()
                             };
-                            labels.push((span, format!("an argument{rendered} is missing")));
+                            let ord = if formal_and_expected_inputs.len() == 1 {
+                                "an".to_string()
+                            } else {
+                                format!("#{}", expected_idx.as_usize() + 1)
+                            };
+                            labels.push((span, format!("{ord} argument{rendered} is missing")));
```

Comment:
similarly

```suggestion
                            labels.push((span, format!("argument{idx} {rendered} is missing")));
```

+ removing the special case `formal_and_expected_inputs.len() == 1` because I don't think it's an improvement: “**An** argument […] is missing“ sounds very non-specific / vague. OTOH “argument #1 […] is missing” is perfectly fine.